addons/website_payment/controllers/portal.py

```diff
@@ -0,0 +1,157 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import json
+import werkzeug
+
+from odoo import http, _
+from odoo.exceptions import ValidationError
+from odoo.http import request
+
+from odoo.addons.payment import utils as payment_utils
+from odoo.addons.payment.controllers import portal as payment_portal
+
+
+class PaymentPortal(payment_portal.PaymentPortal):
+    @http.route(
+        '/payment/pay', type='http', methods=['GET', 'POST'], auth='public', website=True, sitemap=False,
```

Comment:
why POST?

Reply:
because we need to forward a lot of string to this screen that could bypass the url limit length and will be ugly...
The spec is to display on the pay page, the same form that the the donation snippet itself.

https://github.com/odoo/odoo/pull/63133/files#diff-ac50385a414dfae7b74e211e3e1151a2f063a2f82b57e9e8a56db8cc4ebe55edR17

Reply:
As discuss @bso-odoo, after feedback of Olivier, can youn as initially, separate this route from yours.

Thank you

Reply:
Done: routes are separated again